src/main/scala/org/broadinstitute/dsde/firecloud/service/ExportEntitiesByTypeActor.scala

```diff
@@ -1,79 +1,260 @@
-import org.broadinstitute.dsde.firecloud.utils.TSVFormatter
-import spray.http.MediaTypes._
-import spray.http.StatusCodes._
-import spray.http._
+import org.broadinstitute.dsde.firecloud.utils.StreamingActor.{FirstChunk, NextChunk}
+import org.broadinstitute.dsde.firecloud.utils.{StreamingActor, TSVFormatter}
+import org.broadinstitute.dsde.firecloud.{Application, FireCloudConfig, FireCloudExceptionWithErrorReport}
+import org.broadinstitute.dsde.rawls.model._
+import spray.http.{ContentTypes, _}
+import spray.json._
+import spray.routing.RequestContext
 
+import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContext, Future}
-import scala.util.Success
+import scala.language.postfixOps
+
 
 object ExportEntitiesByTypeActor {
+
   sealed trait ExportEntitiesByTypeMessage
-  case class ExportEntities(workspaceNamespace: String, workspaceName: String, filename: String, entityType: String, attributeNames: Option[IndexedSeq[String]]) extends ExportEntitiesByTypeMessage
+
+  case class ExportEntities(ctx: RequestContext, workspaceNamespace: String, workspaceName: String, entityType: String, attributeNames: Option[IndexedSeq[String]]) extends ExportEntitiesByTypeMessage
 
   def props(exportEntitiesByTypeConstructor: UserInfo => ExportEntitiesByTypeActor, userInfo: UserInfo): Props = {
     Props(exportEntitiesByTypeConstructor(userInfo))
   }
 
-  def constructor(app: Application)(userInfo: UserInfo)(implicit executionContext: ExecutionContext) =
-    new ExportEntitiesByTypeActor(app.rawlsDAO, userInfo)
+  def constructor(app: Application, materializer: ActorMaterializer)(userInfo: UserInfo)(implicit executionContext: ExecutionContext) =
+    new ExportEntitiesByTypeActor(app.rawlsDAO, userInfo, materializer)
 }
 
-class ExportEntitiesByTypeActor(val rawlsDAO: RawlsDAO, val userInfo: UserInfo)(implicit protected val executionContext: ExecutionContext) extends Actor with ExportEntitiesByType {
+class ExportEntitiesByTypeActor(val rawlsDAO: RawlsDAO, val argUserInfo: UserInfo, argMaterializer: ActorMaterializer)(implicit protected val executionContext: ExecutionContext) extends Actor with LazyLogging {
+
+  // Requires its own actor context to work with downstream actors: TSVWriterActor and StreamingActor
+  def actorRefFactory: ActorContext = context
+
+  implicit val timeout: Timeout = Timeout(1 minute)
+  implicit val userInfo: UserInfo = argUserInfo
+  implicit val materializer: ActorMaterializer = argMaterializer
+
   override def receive: Receive = {
-    case ExportEntities(workspaceNamespace, workspaceName, filename, entityType, attributeNames) => exportEntities(workspaceNamespace, workspaceName, filename, entityType, attributeNames) pipeTo sender
+    case ExportEntities(ctx, workspaceNamespace, workspaceName, entityType, attributeNames) => streamEntities(ctx, workspaceNamespace, workspaceName, entityType, attributeNames) pipeTo sender
```

Comment:
@davidangb - Let's discuss this when you can. I'd love to simplify the construction here, but I'm completely clueless about how. Rob's suggestion wrt implicits is also relevant here - I don't have the scala chops to make this convoluted construction work any other way than it does now.